Assets/MixedRealityToolkit/Inspectors/Profiles/DataProviderAccessServiceInspector.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    /// <summary>
+    /// Abstract class providing base functionality for data provider management in inspector. Useful for core systems that follow dataprovider access model.
+    /// </summary>
+    public abstract class BaseDataProviderServiceInspector : BaseMixedRealityToolkitConfigurationProfileInspector
+    {
+        protected class ServiceConfigurationProperties
+        {
+            public SerializedProperty Name;
+            public SerializedProperty Type;
+            public SerializedProperty providerProfile;
```

Comment:
Super nit: weird capitalization.

If you want to mach property names why not componentName and componentType as well?